backend/pom.go

```diff
@@ -0,0 +1,138 @@
+package backend
+
+import (
+	"context"
+	"sort"
+	"encoding/xml"
+
+	"github.com/awslabs/yesiscan/interfaces"
+	"github.com/awslabs/yesiscan/util/errwrap"
+	"github.com/awslabs/yesiscan/util/licenses"
+)
+
+const (
+	// PomFilename is the file name used by the pom
+	// files.
+	POMFilename = "pom.xml"
+
+)
+
+//POM or pom.xml files are created from running Java files and they
+//sometimes contain the license of the project.We get the license id(s)
+//by parsing the pom.xml file for its license name(s).
+type Pom struct {
+	Debug bool
+	Logf  func(format string, v ...interface{})
+}
+
+
+type LicenseNames struct{
+	Names []string `xml:"licenses>license>name"`
+}
+
+func (obj *Pom) String() string {
+	return "pom"
+}
+
+func (obj *Pom) ScanData(ctx context.Context, data []byte, info *interfaces.Info) (*interfaces.Result, error) {
+	//this check is taking place with the assumption that 
+	//the file that will be scanned will have to be named "pom.xml"
+	if info.FileInfo.Name() != POMFilename {
+		return nil, nil // skip
+	}
+	if info.FileInfo.IsDir() {
+		return nil, nil // skip
+	}
+	if len(data) == 0 {
+		return nil, nil // skip
+	}
+
+	licenseMap := make(map[string]struct{})
+
+	var list LicenseNames
+	//parsing pom.xml file
+	err := xml.Unmarshal(data, &list)
+	if  err != nil {
+		return nil, errwrap.Wrapf(err, "parse error")
+	}
+	
+	//lid is license id
+	for _, lid := range list.Names {
+		licenseMap[lid] = struct{}{}
+	}
+
+	if len(licenseMap) == 0 {
+		// NOTE: this is NOT the same as interfaces.ErrUnknownLicense
+		// because in this scenario, we're comfortable (ish) the parser
+		// is exhaustive at finding a license with this methodology.
+		// We want to return nil, but we error only if err did
+		// and so normally this returns nil, nil.
+		return nil, errwrap.Wrapf(err, "parse error")
+	}
+
+	ids := []string{}
+	for id := range licenseMap {
+		ids = append(ids, id)
+	}
+	sort.Strings(ids) // deterministic order
+
+	licenseList := []*licenses.License{}
+
+	for _, id := range ids {
+		license := &licenses.License{
+			SPDX: id,
+			// TODO: populate other fields here?
+		}
+
+		// If we find an unknown SPDX ID, we don't want to error,
+		// because that would allow someone to put junk in their code to
+		// prevent us scanning it. Instead, create an invalid license
+		// but return it anyways. If we ever want to check validity, we
+		// know to expect failures.
+		if err := license.Validate(); err != nil {
+			//return nil, err
+			license = &licenses.License{
+				//SPDX: "",
+				Origin: "", // unknown!
+				Custom: id,
+				// TODO: populate other fields here (eg: found license text)
+			}
+		}
+
+		licenseList = append(licenseList, license)
+	}
+
+	result := &interfaces.Result{
+		Licenses:   licenseList,
+		Confidence: 1.0, // TODO: what should we put here?
+	}
+
+	// We perform the strange task of processing any partial results, and
+	// returning some even if we errored, because the spdx code seems to
```

Comment:
Why would we error here?

Reply:
We would error here if the license was invalid.